GraphColoring/ReferenceImplementation.qs

```diff
@@ -95,23 +94,17 @@ namespace Quantum.Kata.GraphColoring {
     operation VertexColoringOracle_Reference (V : Int, edges : (Int, Int)[], colorsRegister : Qubit[], target : Qubit) : Unit is Adj+Ctl {
         let nEdges = Length(edges);
         using (conflicts = Qubit[nEdges]) {
-            for (i in 0 .. nEdges-1) {
-                let (start, end) = edges[i];
-                // Check that endpoints of the edge have different colors:
-                // apply ColorEqualityOracle_Nbit_Reference oracle; if the colors are the same the result will be 1, indicating a conflict
-                ColorEqualityOracle_Nbit_Reference(colorsRegister[start * 2 .. start * 2 + 1], 
-                                                   colorsRegister[end * 2 .. end * 2 + 1], conflicts[i]);
-            }
-
-            // If there are no conflicts (all qubits are in 0 state), the vertex coloring is valid
-            (ControlledOnInt(0, X))(conflicts, target);
-
-            for (i in 0 .. nEdges-1) {
-                let (start, end) = edges[i];
-                // Check that endpoints of the edge have different colors:
-                // apply ColorEqualityOracle_Nbit_Reference oracle; if the colors are the same the result will be 1, indicating a conflict
-                Adjoint ColorEqualityOracle_Nbit_Reference(colorsRegister[start * 2 .. start * 2 + 1], 
-                                                           colorsRegister[end * 2 .. end * 2 + 1], conflicts[i]);
+            within {
+                for (i in 0 .. nEdges-1) {
+                    let (start, end) = edges[i];
```

Comment:
```suggestion
                for (((start, end), conflict) in Zip(edges, conflicts)) {
```

Reply:
I generally prefer not to use Zip, since it complicates reading the code, but I guess this is a sufficiently advanced kata to include some library functions :-)